vm-setup/roles/v1aX_integration_test/filter_plugins/bmh_metadata.py

```diff
@@ -0,0 +1,16 @@
+"""Filter plugins for managing k8s BMH metadata"""
+
+def bmh_nic_names(bmh):
+    """Returns the names of the interfaces of a BareMetalHost as a sorted
+    list.""" 
+    nics = bmh["status"]["hardware"]["nics"]
```

Comment:
would this panic if one of the BMHs did not go through introspection properly and does not have "hardware" key?

Reply:
Yes, it will. Ansible will error and exit. By the time we use this filter in the code, all BMHs are passed inspection and they are provisioned. 

I think this is fine. Ansible's mantra is "fail fast". I would rather we error if something is not right than add safety here.